### PR TITLE
Port additional tests to the new integration test harness

### PR DESCRIPTION
**Pull request by sharwell**
* CSharpGoToDefinition
* CSharpSourceGenerators
* BasicFindReferences
* BasicGoToDefinition

Also includes a few follow-up items from #59783.

Review commit-by-commit may help.

### CODE REVIEW

File: src/VisualStudio/IntegrationTest/New.IntegrationTests/InProcess/LightBulbHelper.cs [hunk 1]
```diff
@@ -62,7 +62,7 @@ public static async Task<IEnumerable<SuggestedActionSet>> WaitForItemsAsync(Test
                 if (e.Status == QuerySuggestedActionCompletionStatus.Completed)
                     tcs.SetResult(e.ActionSets.ToList());
                 else
-                    tcs.SetException(new InvalidOperationException($"Light bulb transitioned to non-complete state: {e.Status}"));
+                    tcs.TrySetException(new InvalidOperationException($"Light bulb transitioned to non-complete state: {e.Status}"));
```
Comment (sharwell) [inline src/VisualStudio/IntegrationTest/New.IntegrationTests/InProcess/LightBulbHelper.cs:65]: 📝 I observed this in an integration test run recently. It races with TrySetCanceled for unexpected dismissal.

File: src/VisualStudio/IntegrationTest/New.IntegrationTests/InProcess/StateResetInProcess.cs [hunk 2]
```diff
@@ -21,13 +23,26 @@ internal partial class StateResetInProcess
         public async Task ResetGlobalOptionsAsync(CancellationToken cancellationToken)
         {
             var globalOptions = await GetComponentModelServiceAsync<IGlobalOptionService>(cancellationToken);
+            ResetOption2(globalOptions, FeatureOnOffOptions.NavigateToDecompiledSources);
             ResetPerLanguageOption(globalOptions, NavigationBarViewOptions.ShowNavigationBar);
+            ResetPerLanguageOption2(globalOptions, VisualStudioNavigationOptions.NavigateToObjectBrowser);
```
Comment (sharwell) [inline src/VisualStudio/IntegrationTest/New.IntegrationTests/InProcess/StateResetInProcess.cs:28]: 📝 Need to update the cleanup to also close the object browser.

File: src/VisualStudio/IntegrationTest/New.IntegrationTests/InProcess/InputInProcess.cs [hunk 3]
```diff
@@ -32,6 +41,14 @@ internal async Task SendAsync(params object[] keys)
             SendKeys.Send(keys);
         }
 
+        internal async Task SendToNavigateToAsync(params object[] keys)
```
Comment (sharwell) [inline src/VisualStudio/IntegrationTest/New.IntegrationTests/InProcess/InputInProcess.cs:44]: 📝 This is super-icky. I want to rewrite `SendKeys` implementation after old integration tests no longer need it.

File: src/VisualStudio/IntegrationTest/New.IntegrationTests/InProcess/WorkspaceInProcess.cs [hunk 4]
```diff
@@ -67,6 +71,23 @@ public async Task WaitForAllAsyncOperationsAsync(string[] featureNames, Cancella
...
+                // Make sure the "priming" operation has started for Nav To
+                var threadingContext = await GetComponentModelServiceAsync<IThreadingContext>(cancellationToken);
+                var asyncListener = listenerProvider.GetListener(FeatureAttribute.NavigateTo);
+                var searchHost = new DefaultNavigateToSearchHost(workspace.CurrentSolution, asyncListener, threadingContext.DisposalToken);
+
+                // Calling DefaultNavigateToSearchHost.IsFullyLoadedAsync starts the fire-and-forget asynchronous
+                // operation to populate the remote host. The call to WaitAllAsync below will wait for that operation to
+                // complete.
+                await searchHost.IsFullyLoadedAsync(cancellationToken);
+            }
```
Comment (sharwell) [inline src/VisualStudio/IntegrationTest/New.IntegrationTests/InProcess/WorkspaceInProcess.cs:89]: 📝 This entire block is also problematic in the old test harness, but it only triggers tests to fail if certain tests run before others. It's a 100% repro when running `InvokeNavigateToForGeneratedFile` by itself.

File: src/VisualStudio/IntegrationTest/New.IntegrationTests/VisualBasic/BasicFindReferences.cs [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+                        Assert.Equal(expected: 4, actual: reference.GetLine());
+                        Assert.Equal(expected: 24, actual: reference.GetColumn());
+                    }
+                });
+        }
+
+        [IdeFact, Trait(Traits.Feature, Traits.Features.FindReferences)]
+        public async Task FindReferencesToSharedField()
+        {
+            //return $"{      hostAddress ?? ""}/{nasTypeId   }/{version ?? ""}";
```
Comment (sharwell) [inline src/VisualStudio/IntegrationTest/New.IntegrationTests/VisualBasic/BasicFindReferences.cs:64]: 📝 This is me testing a customer bug report :facepalm:

**Review by JoeRobich — APPROVED** [review]

File: src/Tools/Source/RunTests/ProcessTestExecutor.cs [hunk 6]
```diff
@@ -75,6 +75,10 @@ void MaybeAddSeparator(char separator = '|')
                 builder.AppendFormat($@" --logger {sep}html;LogFileName={GetResultsFilePath(assemblyInfo, "html")}{sep}");
             }
 
+            // The 25 minute timeout accounts for the fact that VSIX deployment and/or experimental hive reset and
+            // configuration can take significant time (seems to vary from ~10 seconds to ~15 minutes), and the blame
+            // functionality cannot separate this configuration overhead from the first test which will eventually run.
+            // https://github.com/dotnet/roslyn/issues/59851
```
Comment (jasonmalinowski) [inline src/Tools/Source/RunTests/ProcessTestExecutor.cs:81]: Thank you for adding this!

**Pull request merged**